packages/intl-datetimeformat/src/core.ts

```diff
@@ -594,6 +647,42 @@ export function bestFitFormatMatcherScore(
   return score;
 }
 
+function dateTimeStyleFormat(
+  dateStyle?: DateTimeFormatOptions['dateStyle'],
+  timeStyle?: DateTimeFormatOptions['timeStyle'],
+  // @ts-ignore
+  formats: CalendarFormats
```

Comment:
why is this `ts-ignore`d?

Reply:
ts does not allow required params after optional
```
A required parameter cannot follow an optional parameter.ts(1016)
```
However, i just discovered you can do `DateTimeFormatOptions['dateStyle'] | undefined` and it is not considered an optional parameter